From 8402c24b7852f19da2a48879512baf295e9df3e7 Mon Sep 17 00:00:00 2001
From: Liming Wang <liming.wang@windriver.com>
Date: Fri, 30 Nov 2012 14:54:44 +0800
Subject: [PATCH] arm/zynq: Update device tree for xilinx qemu

Update device tree for xilinx qemu:

 - remove unsupported nor flash
 - add qspi flash support

Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 arch/arm/boot/dts/zc702-qemu.dtsi |   51 +++++++++++++++++++++++++++++++------
 1 file changed, 43 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/zc702-qemu.dtsi b/arch/arm/boot/dts/zc702-qemu.dtsi
index bd40ad2..88a0ac8 100644
--- a/arch/arm/boot/dts/zc702-qemu.dtsi
+++ b/arch/arm/boot/dts/zc702-qemu.dtsi
@@ -118,15 +118,50 @@
 			clock-frequency = <33333000>;
 		};
 
-		nor@e2000000 {
-			compatible = "cfi-flash";
-			bank-width = <1>;
-			reg = <0xE2000000 0x1000000>;
+		qspi@e000d000 {
+			compatible = "xlnx,ps7-qspi-1.00.a";
+			reg = <0xe000d000 0x1000>;
 			#address-cells = <1>;
-			#size-cells = <1>;
-			partition@0 {
-				label = "testnor";
-				reg = <0 0x400000>;
+			#size-cells = <0>;
+			interrupts = <0x00 0x13 0x04>;
+			speed-hz = <200000000>;
+			bus-num = <0x1>;
+			num-chip-select = <1>;
+			is-dual = <0x0>;
+			flash@0 {
+				compatible = "n25q128";
+				reg = <0x0>;
+				spi-max-frequency = <50000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+                                partition@qspi-fsbl {
+                                        label = "qspi-fsbl";
+                                        reg = <0x0 0x80000>;
+                                };
+                                partition@qspi-u-boot {
+                                        label = "qspi-u-boot";
+                                        reg = <0x80000 0x80000>;
+                                };
+                                partition@qspi-linux {
+                                        label = "qspi-linux";
+                                        reg = <0x100000 0x500000>;
+                                };
+                                partition@qspi-device-tree {
+                                        label = "qspi-device-tree";
+                                        reg = <0x600000 0x20000>;
+                                };
+                                partition@qspi-user {
+                                        label = "qspi-user";
+                                        reg = <0x620000 0xE0000>;
+                                };
+                                partition@qspi-scratch {
+                                        label = "qspi-scratch";
+                                        reg = <0x700000 0x100000>;
+                                };
+                                partition@qspi-rootfs {
+                                        label = "qspi-rootfs";
+                                        reg = <0x800000 0x800000>;
+                                };
 			};
 		};
 
-- 
1.7.9.5

